themes/origins/assets/component-write-review.js

```diff
@@ -0,0 +1,119 @@
+    this.form = this.querySelector("[data-review-form]");
+    this.imageInput = this.form.querySelector("input[name='images']");
+    this.imagesContainer = this.querySelector("[data-images-container]");
+
+    this.submitButton = this.querySelector("[data-submit]");
+  }
+
+  connectedCallback() {
+    this._render();
+  }
+
+  _render() {
+    this.form.addEventListener("submit", this.submitReview.bind(this));
+    this.imageInput.addEventListener("change", this.onUploadImage.bind(this));
+  }
+
+  async submitReview(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+
+    if (Number(formData.get("ratings")) === 0) {
+      const message = window.errorStrings.rating_is_required;
+      toast.show(message, "error");
+
+      return;
+    }
+
+    const payload = {
+      images: this.images,
+      first_name: formData.get("first-name"),
+      last_name: formData.get("last-name"),
+      ratings: +formData.get("ratings"),
+      content: formData.get("content"),
+      email: formData.get("email"),
+    };
+    this.toggleSubmitButton(true);
+
+    try {
+      const response = await youcanjs.product.submitReview(this.productId, payload);
+      this.resetForm();
+
+      toast.show(response.detail, "success");
+    } catch (error) {
+      console.error(error);
+
+      toast.show(error.message, "error");
+    } finally {
+      this.toggleSubmitButton(false);
+    }
+  }
+
+  onUploadImage(event) {
+    const files = Array.from(event.target.files);
+    if (files.length < 1) return;
+
+    files.forEach((file) => {
+      const fileSizeInKB = file.size / WriteReview.BYTES_IN_KB;
+      const fileSizeInMB = fileSizeInKB / WriteReview.KB_IN_MB;
+
+      if (fileSizeInMB <= WriteReview.MAXIMUM_FILE_SIZE) {
+        const reader = new FileReader();
+
+        reader.onload = () => {
+          const base64 = reader.result;
+          this.addImage(base64);
+        };
+        reader.readAsDataURL(file);
+      } else {
+        const message = window.errorStrings.large_file;
+        toast.show(message.replace("[file]", `"${file.name}"`), "error");
+      }
+    });
+  }
+
+  addImage(source) {
+    this.images.push(source);
+    const template = this.querySelector("[data-image]");
+    const image = template.content.cloneNode(true);
+    const imgElement = image.querySelector("img");
+    const removeButton = image.querySelector("[data-remove-image]");
+
+    imgElement.src = source;
+    this.imagesContainer.appendChild(image);
+
+    removeButton.addEventListener("click", (event) =>
+      this.removeImage(source, event.currentTarget.parentElement),
+    );
+  }
+
+  removeImage(source, element) {
+    element?.remove();
+    this.images = this.images.filter((image) => image !== source);
+  }
+
+  resetForm() {
+    this.images = [];
+    this.imagesContainer.innerHTML = "";
+    this.form.reset();
+    this.modal.close();
+  }
+
+  toggleSubmitButton(is_loading) {
```

Comment:
Toggle is a little vague, it gives the impression that you're toggling the actual button (visible/invisible)

```suggestion
  setSubmitButtonIsLoading(is_loading) {
```